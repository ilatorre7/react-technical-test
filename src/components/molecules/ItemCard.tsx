import '../../styles/ItemCard.css'
import { formatPrice } from '../../utils'
import { IItem } from '../types'
import { Button } from '../atoms/Button'


export const ItemCard = ({ id, title, description, image, detail, itemRef, onClick } : IItem & { onClick: ({ id }: { id: number }) => void, itemRef: ((node: HTMLElement) => void) | null } ) => {
  const handleClick = () => {
    onClick({ id })
  }

  return (
    <article className='card-container' ref={itemRef}>
      <div className='card-content'>
        <img className='card-image' src={image} alt={title} />
        <div className='card-info'>
          <header className='card-header'>
            <h3 className='card-title'>{title}</h3>
            <h5>{detail.year}</h5>
          </header>
          <p className='card-description'>{description}</p>
          <h4 className='card-price'>{formatPrice(detail.price)}</h4>
          <Button label='Detalles' onClick={handleClick} />
        </div>
      </div>
    </article>
  )
}