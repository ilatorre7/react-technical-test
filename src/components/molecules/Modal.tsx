import { IItem } from "../types"
import '../../styles/Modal.css'
import { formatPrice } from "../../utils"
import cross from '../../assets/x.svg';

export const Modal = ({ item, setOpen } : { setOpen: React.Dispatch<React.SetStateAction<boolean>>, item: IItem | undefined }) => {

  return (
    <>
      <div className='background' onClick={() => setOpen(false)}>
      </div>
      <div className='modal-container'>
        <div className='modal-content'>
          <img className="modal-close" src={cross} alt='Close modal' onClick={() => setOpen(false)} />
          <img className='modal-image' src={item?.image} alt={item?.title}/>
          <div className='modal-info'>
            <h1>{item?.title}</h1>
            <h3>{item?.detail.year}</h3>
            <p className='modal-description'>{item?.description}</p>
            <ul className='modal-list'>
              <li>Marca: {item?.detail.brand}</li>
              <li>Colorway: {item?.detail.colorway}</li>
            </ul>
            <h2 className='modal-price'>{formatPrice(item?.detail.price || '')}</h2>
          </div>
        </div>
      </div>
    </>
  )
}