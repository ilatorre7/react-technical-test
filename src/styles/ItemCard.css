.card-container {
  overflow: hidden;
  border: 1px solid #ccc;
  border-radius: 16px;
  background-color: #fff;
}

.card-content {
  display: flex;
  align-items: center;
  gap: 32px;
}

.card-image {
  width: 320px;
  height: 320px;
  object-fit: cover;
}

.card-header {
  display: flex;
  flex-direction: column;
  gap:8px;
  align-items: baseline;
}

.card-info {
  padding: 32px 16px;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  gap: 16px;
}

.card-title {
  font-size: 20px;
}

.card-price {
  font-size: 32px;
  font-weight: 700;
}

.card-description {
  color: var(--main-text-color);
}

@media (max-width: 640px) {
  .card-content {
    flex-direction: column;
    width: 100%;
  }

  .card-content {
    gap: 0px;
  }

  .card-image {
    width: 100%;
  }

  .card-info {
    padding: 16px;
  }

}