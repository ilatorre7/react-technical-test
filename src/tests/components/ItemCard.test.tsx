import { describe, it, expect } from 'vitest'
import { render } from '@testing-library/react';
import { ItemCard } from '../../components/molecules/ItemCard';

describe('<ItemCard />', () => {
  const expectedCard = {
    id: 1,
    title: "Jordan 1 Low OG",
    image: "https://nikearprod.vtexassets.com/arquivos/ids/880101-800-800?width=800&height=800&aspect=true",
    description: "Comienza el Año del Dragón con estilo. Este elegante AJ1 tiene superposiciones escamosas en verde petróleo inspiradas en la única criatura mitológica del zodíaco chino.",
    detail: {
      price: "299999",
      year: 2023,
      brand: "Jordan",
      colorway: "Verde petróleo/Metallic Gold"
    }
  }

  it('Should render a card with the correct values', () => {
    const { container } = render(<ItemCard {...expectedCard } />)
    expect(container).toMatchSnapshot()
  })
})