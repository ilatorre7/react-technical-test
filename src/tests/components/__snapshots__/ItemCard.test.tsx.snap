// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ItemCard /> > Should render a card with the correct values 1`] = `
<div>
  <article
    class="card-container"
  >
    <div
      class="card-content"
    >
      <img
        alt="Jordan 1 Low OG"
        class="card-image"
        src="https://nikearprod.vtexassets.com/arquivos/ids/880101-800-800?width=800&height=800&aspect=true"
      />
      <div
        class="card-info"
      >
        <header
          class="card-header"
        >
          <h3
            class="card-title"
          >
            Jordan 1 Low OG
          </h3>
          <h4>
            2023
          </h4>
        </header>
        <p>
          Comienza el Año del Dragón con estilo. Este elegante AJ1 tiene superposiciones escamosas en verde petróleo inspiradas en la única criatura mitológica del zodíaco chino.
        </p>
        <h4
          class="card-price"
        >
          $299.999
        </h4>
        <button
          class="button-container"
        >
          Detalles
        </button>
      </div>
    </div>
  </article>
</div>
`;
